src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http3Connection.cs

```diff
@@ -130,6 +134,10 @@ private void CheckForShutdown()
             if (_connection != null)
             {
                 // Close the QuicConnection in the background.
+                if (WTManager is not null)
+                {
+                    WTManager!.Dispose();
+                }
```

Comment:
@CarnaViire Yes, because if it is before, the `CheckForShutdown` function ca be called multiple times. Also I think it is more safe for the session to be disposed before the Quic Connection is closed